stan/math/opencl/rev/copy.hpp

```diff
@@ -53,6 +53,31 @@ inline var_value<matrix_cl<value_type_t<T>>> to_matrix_cl(
                                                                  a.vi_->adj_);
 }
 
+/** \ingroup opencl
+ * Copies the source std::vector of vars to a destination var that has
+ * data stored on the OpenCL device.
+ *
+ * @tparam T type of the std::vector
+ * @param a source Eigen matrix
+ * @return var with a copy of the data on the OpenCL device
+ */
+template <typename T, require_stan_scalar_t<T>* = nullptr>
+inline var_value<matrix_cl<value_type_t<T>>> to_matrix_cl(
+    std::vector<var_value<T>>& a) {
+  return new internal::op_copy_to_cl_vari<decltype(
```

Comment:
This is not ok. The vector can get destructed before the chain is called. Instead you should just call `to_matrix_cl` (the last overload in this file) on the map. That will take care of storing the varis in the arena.

Reply:
Thanks. Done.